File: src/main/java/com/jessica/student/Student.java
package com.jessica.student;

public class Student {
    String id;
    String name;
    int english;
    int math;
    static int pass=60;
    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest(){
        return (english>math)?english:math;
    }
    public void grading(){
        int average =getAverage();
        System.out.println(name+"\t"+english+"\t"+math+"\t"+(getAverage()>=60?"PASS":"FALIED"));
        char grading ='F';
        switch(average/10){
            case 10:
            case 9:
                grading='A';
                break;
            case 8:
                grading='B';
                break;
            case 7:
                grading='C';
                break;
            case 6:
                grading='D';
                break;
            default:
                grading='F';
        }

    }
    public int getAverage(){
        return (english+math)/2;
    }
}

File: src/main/kotlin/com/student/Student.kt
package com.kotlin

open class Student(var name :String?,var math: Int,var english:Int) {
    companion object{
        val pass=60
        fun test(){
            println("testing")
        }
    }
class GraduateStudent(name: String?,english: Int,math: Int,thesis:Int):Student(name,english,math){
    companion object{
        var pass=70
    }
}

    open fun print(){
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}")
    }
   protected fun grading()= when (getAverage()) {
            in 90..100 -> 'A'
            in 80..89 -> 'B'
            in 70..79 -> 'C'
            in 60..69 -> 'D'
            else -> 'F'
    }

    fun passOrFailed()=if(getAverage()>= pass)"PASS" else "FAILED"
    fun getAverage()=(math+english)/2

    fun nameCheck(){
        print(name?.length)
    }
    fun highest()= if(english>math)english else math

}


fun main(args: Array<String>) {
    val stu1= Student("Hank",90,80)
    val stu2= Student("A",60,50)
    val stu3= Student("B",10,10)
    stu1.print()
    stu2.print()
    stu3.print()
//    println("High score:${stu.highest()}")\
}


